ludwig/data/dataset/ray.py

```diff
@@ -250,7 +205,15 @@ def __init__(
         self.dataset_shard = dataset_shard
         self.features = features
         self.training_set_metadata = training_set_metadata
-        self.epoch_iter = dataset_shard.iter_epochs()
+        self.create_epoch_iter()
+
+    def create_epoch_iter(self) -> None:
+        if isinstance(self.dataset_shard, DatasetPipeline):
+            self.epoch_iter = self.dataset_shard.iter_epochs()
+        else:
+            # Dataset shard is a Ray Dataset object
+            # Convert Ray Dataset to a DatasetPipeline object before enabling epoch iteration
+            self.epoch_iter = self.dataset_shard.repeat().iter_epochs()
```

Comment:
How does this work with shuffling / windowing from earlier in the pipeline? If we're repeating over the shard, seems we will never perform any shuffling. Am I missing something?

Reply:
You are right! 

What I should've called out in the comments here is that `self.dataset_shard` will only be a `ray.data.Dataset` object when we're doing auto batch size tuning or learning rate tuning, in which case we don't actually need to shuffle or window. In all other cases (regular training, prediction etc), it will be a DatasetPipeline object which will allow windowing/shuffles, etc through the DatasetConfig.

Will call this out in a better comment.